src/workloads/execution/UpdateLargeDocuments.yml

```diff
@@ -0,0 +1,83 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: >
+  This workload measures the performance of large document updates in a replica set. First, it
+  inserts documents with large embedded objects, containing 100k fields each. Then it runs a series
+  of simple updates. Each update is setting only one field and has write concern "majority". The
+  output of this workload is ops/sec, indicating how many of these simple updates were finished in a
+  period of time.
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 10
+
+Actors:
+
+# Phases:
+# 1. Insert data with large documents roughly resembling buckets for time-series collection
+# 2. Fsync
+# 3. Perform very simple update
+
+- Name: InsertData
+  Type: Loader
+  # Loader doesn't support multiple threads for a single collection.
+  # TODO: TIG-2938 Use multiple threads to speedup insertion.
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: &DB test
+    Threads: 1
+    CollectionCount: 1  # Collection name will be Collection0, this is not configurable.
+    DocumentCount: &DocumentCount 100
+    BatchSize: 100
+    Document:
+      _id: {^Inc: {start: 0}}
+      data: {^Object: {
+        key: {^FastRandomString: {length: { ^RandomInt: { min: 1, max: 10 }}}},
+        value: {^Object: {
+          key: {^FastRandomString: {length: { ^RandomInt: { min: 1, max: 3 }}}},
+          value: { ^RandomInt: { min: 1, max: 10 }},
+          number: 100
+        }},
+        number: 1000
+      }}
+      a: 0
+  - &Nop {Nop: true}
+  - *Nop
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *DB
+  Phases:
+  - *Nop
+  - Repeat: 1
+  - *Nop
+
+- Name: PerformUpdates
+  Type: CrudActor
+  Database: *DB
+  ClientName: Default
+  Phases:
+  - *Nop
+  - *Nop
+  - Repeat: 1000
+    Threads: 50
+    Duration: 3 minutes
+    Collection: Collection0
+    Operations:
+    - OperationName: updateOne
+      OperationCommand:
+        Filter: {_id: { ^RandomInt: { min: 0, max: *DocumentCount } }}
+        Update: {$set: {a: { ^RandomInt: { min: 0, max: 10000000 }}}}
```

Comment:
Other questions:

- should there be a case where we update random strings or a random selection of different types (or at least non fixed size types like ints)?
- what about $unset ?
- what about different sized deltas (updating multiple fields)?

This ticket is specific to large documents,  do we also need a workload for different document sizes?

Reply:
> should there be a case where we update random strings or a random selection of different types (or at least non fixed size types like ints)?
> what about different sized deltas (updating multiple fields)?

In this workload, we would like to stress the extra overhead for the update operation. The update operation should be simple, so that we will be able to measure other parts of request execution (such as result document validation).

> what about $unset?

I am not sure we will get extra coverage from adding test with $unset operator. We are really interested in the non-update parts of the execution.